tests/test_fields.py

```diff
@@ -475,6 +483,14 @@ def test_fails_on_value_not_matching_pattern(self):
         self.assertRaises(errors.FieldValueError, field_format.validated, "hang")
 
 
+class PublicFieldFunctionTest(unittest.TestCase):
```

Comment:
Migrate this test to the `test_interface` and make it a variant of `test_fails_on_broken_field_name`. As discussed the error message `field name must begin with a lower-case letter but is ...` should actually be `...must be an ASCII letter...`.